features/nvstore/source/nvstore.h

```diff
@@ -159,22 +159,14 @@ class NVStore : private mbed::NonCopyable<NVStore> {
     int set(uint16_t key, uint16_t buf_size, const void *buf);
 
     /**
-     * @brief Programs one item of data on Flash, allocating a key.
+     * @brief Allocate a free key (to be used later in a set operation).
      *
      * @param[out] key                  Returned key of stored item.
-     * @param[in]  buf_size             Item size in bytes.
-     * @param[in]  buf                  Buffer containing data.
      *
-     * @returns NVSTORE_SUCCESS           Value was successfully written on Flash.
-     *          NVSTORE_WRITE_ERROR       Physical error writing data.
-     *          NVSTORE_BAD_VALUE         Bad value in any of the parameters.
-     *          NVSTORE_FLASH_AREA_TOO_SMALL
-     *                                    Not enough space in Flash area.
-     *          NVSTORE_ALREADY_EXISTS    Item set with write once API already exists.
      *          NVSTORE_NO_FREE_KEY       Couldn't allocate a key for this call.
      *
      */
-    int set_alloc_key(uint16_t &key, uint16_t buf_size, const void *buf);
+    int allocate_key(uint16_t &key);
```

Comment:
This is a breaking change.

Reply:
It is indeed, but the replaced function (`set_alloc_key`) is a new one that was implemented after the 5.8 release. It's still not even documented. Figured that the API we had defined was wrong in that case, hence the change. Can add the "old" API back, but that that would be just dead code.

Reply:
This PR contains a breaking change, so marked for 5.9.0.

Reply:
That's OK. It is mainly used for StorageLite feature, which should be out in 5.9.0.